dotcom-rendering/src/lib/useCommentCount.ts

```diff
@@ -0,0 +1,33 @@
+import { isNonNullable } from '@guardian/libs';
+import { isServer } from './isServer';
+import { useApi } from './useApi';
+
+type CommentCounts = Record<string, number>;
+
+const getInitialSetFromDOMAttribute = (attribute: string) =>
+	[...document.querySelectorAll(`[${attribute}]`)]
+		.map((element) => element.getAttribute(attribute))
+		.filter(isNonNullable);
+
+const uniqueDiscussionIds = new Set<string>(
+	isServer ? [] : getInitialSetFromDOMAttribute('data-discussion-id'),
```

Comment:
Will this run on the server because it will evaluate at a module level? If this method should only run on the client perhaps it is better to make it a function  or inline it so it resolves as part of the component instantiation.

Reply:
We want this set to be [shared across all usages of the hook](https://react.dev/learn/reusing-logic-with-custom-hooks#custom-hooks-let-you-share-stateful-logic-not-state-itself), and ideally only initilised once. I’m not too clear on what hooks running on the server mean.

Reply:
Non-blocking I will catch up with you to understand this feature better.